### PR TITLE
pointreg app-to-gtest

### PR DESCRIPTION
**Pull request by tgiroux**
## Description

Tests utilize ThreeImageNetwork with a gradient added to each cube. I used the Makefile tests as a template for coverage.

## How Has This Been Tested?

This is a test
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [x] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/tests/FunctionalTestsPointreg.cpp [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+  // Populate cubes
+  LineManager line(*cube1);
+  double pixelValue = 0.0;
+  for(line.begin(); !line.end(); line++) {
+    for(int i = 0; i < line.size(); i++) {
+      line[i] = pixelValue++;
+    }
+    cube1->write(line);
+    cube2->write(line);
+    cube3->write(line);
```
Comment (Kelvinrr) [inline isis/tests/FunctionalTestsPointreg.cpp]: I would just add this as part of the fixture

File: isis/tests/FunctionalTestsPointreg.cpp [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+  UserInterface optionsE(APP_XML, argsE);  // Output unmeasured, register ignored
+  Pvl logA, logB, logC, logD, logE;
+
+  try {
+    pointreg(optionsA, &logA);
+    pointreg(optionsB, &logB);
+    pointreg(optionsC, &logC);
+    pointreg(optionsD, &logD);
+    pointreg(optionsE, &logE);
+  }
```
Comment (Kelvinrr) [inline isis/tests/FunctionalTestsPointreg.cpp]: usually when a Makefile has a few tests in it, they are separate tests that they felt needed to be categorized. But in the gtests, we just separate them to individual tests. I personally advocate splitting these up into four tests with exception to potentially redundant tests.

File: isis/tests/FunctionalTestsPointreg.cpp [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+                            "onet=" + outNetPath,
+                            "outputignored=no",
+                            "points=nonignored" };
+  QVector<QString> argsE = { "fromlist=" + cubeListFile,
+                            "cnet=data/threeImageNetwork/controlnetwork.net",
+                            "deffile=data/threeImageNetwork/autoRegTemplate.def",
+                            "flatfile=" + flatFilePath,
+                            "onet=" + outNetPath,
+                            "outputignored=no",
+                            "points=ignored" };
```
Comment (Kelvinrr) [inline isis/tests/FunctionalTestsPointreg.cpp]: some of these options look the same? If they are redundant, I would just remove them.

File: isis/tests/FunctionalTestsPointreg.cpp [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+                            "outputfailed=no",
+                            "outputignored=no",
+                            "points=ignored" };
+  QVector<QString> argsB = { "fromlist=" + cubeListFile,
+                            "cnet=data/threeImageNetwork/controlnetwork.net",
+                            "deffile=data/threeImageNetwork/autoRegTemplate.def",
+                            "flatfile=" + flatFilePath,
+                            "onet=" + outNetPath,
+                            "outputfailed=no",
+                            "outputignored=no" };
```
Comment (Kelvinrr) [inline isis/tests/FunctionalTestsPointreg.cpp]: same with splitting them up here.

File: isis/tests/FunctionalTestsPointreg.cpp [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+
+  Pvl log;
+  QTemporaryDir prefix;
+  QString flatFilePath = prefix.path() + "/flatfile.csv";
+  QString outNetPath = prefix.path() + "/outNet.net";
+  QVector<QString> args = { "fromlist=" + cubeListFile,
+                            "cnet=data/threeImageNetwork/controlnetwork.net",
+                            "deffile=data/threeImageNetwork/autoRegTemplate.def",
+                            "flatfile=" + flatFilePath,
+                            "onet=" + outNetPath };
```
Comment (Kelvinrr) [inline isis/tests/FunctionalTestsPointreg.cpp]: I think the fixture has a var for the cnet path, I would use that and add the template to a variable as well. It'll make things easier to move around in the future.

**Review by Kelvinrr — COMMENTED** [review]
some suggestions inline

File: isis/tests/FunctionalTestsPointreg.cpp (hunk 2 above)
Comment (tgiroux) [reply isis/tests/FunctionalTestsPointreg.cpp]: Yeah I agree, best to stick with our convention and this way of testing these cases is pretty ugly anyways.

**Comment by tgiroux** [comment]
There's a merge conflict Fixtures.cpp from the findfeatures PR where Austin already added data to the cubes in ThreeImageNetwork. The trouble is the data is a bit different than what I've been using, which is causing some differing output from my test cases. Gonna look it back over and add another commit to make my tests work with the new cube data.

**Comment by tgiroux** [comment]
Should be good for a rereview now.

**Review by Kelvinrr — DISMISSED** [review]

**Comment by Kelvinrr** [comment]
* after conflict resolution

**Comment by AustinSanders** [comment]
@Kelvinrr The conflict is super simple.  Can I just use the web editor to resolve conflicts and get this merged?  It's just:

```
    networkFile = "data/threeImageNetwork/controlnetwork.net";

    network = new ControlNet();
<<<<<<< pointreg_test
    network->ReadControl(networkFile);
  }
=======
    network->ReadControl("data/threeImageNetwork/controlnetwork.net");
>>>>>>> dev

```

**Comment by Kelvinrr** [comment]
@AustinSanders I recall Tim mentioning wanting to fix these but since it so simple and the day is almost over I just ended up doing it like you mentioned. 

Doesn't seem to be deploying though? Had to run manually.

**Comment by acpaquette** [comment]
@Kelvinrr This is busted now. Looks like some type defs got removed from things. Maybe pull and build locally?

**Comment by Kelvinrr** [comment]
@acpaquette yeah, I'll probably finish this and rebecca's.

**Review by Kelvinrr — APPROVED** [review]

**Pull request merged**